### DISCUSSION TITLE
config file, set the parameters according to the respective server

### DISCUSSION THREAD

**Discussion by sky186** [milvus-io/milvus#4638] [Q&A and General discussion]
1、一次插入到 milvus 表中的数据  不超过 256M，是否是如下位置自定义修改大小，
buffer_size必须在[64，4096]（MB）范围内
wal:
  enable: true
  recovery_error_ignore: false
  buffer_size: 256MB
  path: /var/lib/milvus/wal

2、建立索引搜索阶段 **_INDEX_FILE_SIZE** = 1024  # max file size of stored index

https://www.milvus.io/cn/blogs/2020-2-16-api-setting.md/  参数文档介绍是设置1024
这里 设置64 等小一点的值（2048向量），正常不报错，**这里设置的大小对 精度速度的影响，如何设置一个合适的值？**

    WARN: increase temp memory to avoid cudaMalloc, or decrease query/add size (alloc 131072 B, highwater 268977408 B)
    Faiss assertion 'err == cudaSuccess' failed in void faiss::gpu::freeMemorySpace(faiss::gpu::MemorySpace, void*) at gpu/utils/MemorySpace.cpp:72; details: Failed to cudaFree pointer 0x7efdec000000 (error 1 invalid argument)

3、congfig文件 cpu ,gpu 大小容量如何根据各自电脑设置，对搜索的影响，根据不同规模数据会不同设置吗
例如四卡 v100 (32G) 显卡 
 cpu  和 gpu 几个 size 设置的 标准是什么， （gpu 多卡，也许有其他人的任务占用，会怎么样）

cache:
  cache_size: 128GB
  insert_buffer_size: 4GB
  preload_collection:

gpu:
  enable: true
  cache_size: 4GB
  gpu_search_threshold: 1000
  search_devices:
    - gpu0
  build_index_devices:
    - gpu0

**Comment by shiyu22** [comment]
1. The amount of data cannot exceed 256M when inserting in bulk, this is a system setting that cannot be modified. About the `buffer_size` info, you can refer to https://www.milvus.io/docs/v0.10.5/milvus_config.md.
2. For the recommended `index_file_size` values, you can refer to [the article](https://www.milvus.io/cn/blogs/2020-2-16-api-setting.md).
3. For the config of gpu/cpu, you can refer to [this article](https://www.milvus.io/cn/blogs/2020-2-19-milvus-config.md).
